.github/workflows/deploy-docs.yml

```diff
@@ -0,0 +1,47 @@
+# This workflow builds and deploys Sphinx documentation to the gh-pages branch
+# It builds when a tag is _released_ (meaning not a prerelease).
+
+name: Build and Deploy Sphinx Docs
+
+
+on:
+  release:
+    types: [released]
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+# follows https://github.com/marketplace/actions/deploy-to-github-pages
+jobs:
+  build-and-deploy:
+    runs-on: ubuntu-latest
+
+    steps:
+      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+      - uses: actions/checkout@v2
+        with:
+          persist-credentials: false
+
+      # Set up the python environment.
+      - name: Setup Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: '3.x'
+      
+      # Set up dependencies.
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install sphinx sphinx_rtd_theme
+
+      # Build the documentation
+      - name: Build Docs
+        run: |
+          make documentation
+      
+      # Use https://github.com/marketplace/actions/deploy-to-github-pages to deploy docs.
+      - name: Deploy Docs
+        uses: JamesIves/github-pages-deploy-action@3.7.1
+        with:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
Is this something that is automatically available to all GH actions? i.e. we don't need to generate something and add it as a repo secret?

Reply:
According to the GH docs, it is automatically created. https://docs.github.com/en/free-pro-team@latest/actions/reference/authentication-in-a-workflow#about-the-github_token-secret